File: test/benchmark/cases/std_vector_string/case.cpp
#include <cstdint>
#include <cstdlib>
#include <string>
#include <vector>

auto main() -> int
{
    auto const* raw = std::getenv("CP_BENCH_INPUT");
    if(raw == nullptr) {
        return 1;
    }

    auto const count = std::atoi(raw);
    auto values = std::vector<int>{};
    auto text = std::string{};
    values.reserve(static_cast<std::size_t>(count));
    text.reserve(static_cast<std::size_t>(count));

    auto index = 0;
    while(index < count) {
        values.push_back((index * 13 + 5) % 997);
        text.push_back(static_cast<char>('a' + (index % 26)));
        index += 1;
    }

    auto total = static_cast<std::int64_t>(text.size());
    auto cursor = std::size_t{ 0 };
    while(cursor < values.size()) {
        total += values[cursor];
        cursor += 1;
    }
    return static_cast<int>(total % 251);
}

File: test/benchmark/cases/std_vector_string/case.java
import java.util.ArrayList;

class Main {
    public static void main(String[] args) {
        var raw = System.getenv("CP_BENCH_INPUT");
        if(raw == null) {
            System.exit(1);
        }

        var count = Integer.parseInt(raw);
        var values = new ArrayList<Integer>(count);
        var text = new StringBuilder(count);

        var index = 0;
        while(index < count) {
            values.add((index * 13 + 5) % 997);
            text.append((char)('a' + (index % 26)));
            index += 1;
        }

        var total = (long)text.length();
        var cursor = 0;
        while(cursor < values.size()) {
            total += values.get(cursor);
            cursor += 1;
        }
        System.exit((int)(total % 251));
    }
}
